### PR TITLE
Add support of basic data types byte, short, binary and add data type testing

### PR DESCRIPTION
**Pull request by chloe-zh**
*Issue #, if available:*

*Description of changes:

- Added support of basic data types including `BYTE`, `SHORT` (short type was already partly implemented before this PR), and es type `BINARY`
- Enabled `BYTE` and `SHORT` in arithmetic operations and some mathematical functions
- Added support of `BINARY` type in JDBC driver
- Added integration test for all data types we have supported in the core engine

Ref: https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-data-types.html

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/DataTypeIT.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+public class DataTypeIT extends PPLIntegTestCase {
+
+  @Override
+  public void init() throws IOException {
+    loadIndex(DATA_TYPE_NUMERIC);
+    loadIndex(DATA_TYPE_NONNUMERIC);
+  }
+
+  @Test
+  public void test_numeric_data_types() throws IOException {
```
Comment (penghuo) [inline integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/DataTypeIT.java:38]: do we need to verify the row?

File: elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/data/value/ElasticsearchExprBinaryValue.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+package com.amazon.opendistroforelasticsearch.sql.elasticsearch.data.value;
+
+import com.amazon.opendistroforelasticsearch.sql.data.model.AbstractExprValue;
+import com.amazon.opendistroforelasticsearch.sql.data.model.ExprValue;
+import com.amazon.opendistroforelasticsearch.sql.data.type.ExprType;
+import com.amazon.opendistroforelasticsearch.sql.elasticsearch.data.type.ElasticsearchDataType;
+import lombok.EqualsAndHashCode;
+
+@EqualsAndHashCode(callSuper = false)
+public class ElasticsearchExprBinaryValue extends AbstractExprValue {
```
Comment (penghuo) [inline elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/data/value/ElasticsearchExprBinaryValue.java:25]: doc?

File: core/src/main/java/com/amazon/opendistroforelasticsearch/sql/data/model/ExprByteValue.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.data.model;
+
+import com.amazon.opendistroforelasticsearch.sql.data.type.ExprCoreType;
+import com.amazon.opendistroforelasticsearch.sql.data.type.ExprType;
+
+/**
+ * Expression Short Value.
```
Comment (penghuo) [inline core/src/main/java/com/amazon/opendistroforelasticsearch/sql/data/model/ExprByteValue.java:22]: Short -> Byte

**Review by penghuo — COMMENTED** [review]
could you also change the doc. https://github.com/opendistro-for-elasticsearch/sql/blob/develop/docs/user/general/datatypes.rst

File: elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/data/value/ElasticsearchExprBinaryValue.java (hunk 2 above)
Comment (chloe-zh) [reply elasticsearch/src/main/java/com/amazon/opendistroforelasticsearch/sql/elasticsearch/data/value/ElasticsearchExprBinaryValue.java:29]: added

**Review by penghuo — APPROVED** [review]
Thanks for the change!

**Review by anirudha — APPROVED** [review]

**Pull request merged**